### ISSUE TITLE
USI Reaktor with Kerbalism 

### ISSUE THREAD

**Issue by Plerx2493** [jarosm/KSP-BonVoyage#35]
When you have installed Kerbalism, the USI reactors are no longer recognized. This is because Kerbalism replaces the ModuleResourceConverter with a ProcessController module:

MODULE
	{
		name = ProcessController
		resource = _Nukereactor
		title = Fission reactor
		capacity = 1
	}
	
The capacity is 1/10th of the generated EC.

**Comment by Plerx2493** [comment]
I wrote an improvement and I'm currently testing it

**Comment by gotmachine** [comment]
This is a wider issue than just USI reactors.
BV is using its own implementations to get the EC generation / EC state of the vessel in the background, by checking against the stock ProtoModules / ProtoResources.
Kerbalism has a full fledged background resource simulation, so all this isn't needed at all. 
The current Kerbalism support adds EC consumption through Kerbalism to the BV controller :
https://github.com/jarosm/KSP-BonVoyage/blob/81ba8c575d53e9475ee07ca906eb10b2b635f884/BonVoyage/BVController.cs#L613-L617
and 
https://github.com/jarosm/KSP-BonVoyage/blob/81ba8c575d53e9475ee07ca906eb10b2b635f884/BonVoyage/BonVoyageModule.cs#L320-L348

However, it still use its stock checks to determine the available EC, which doesn't work well with Kerbalism.
Fixing it would require to not do these :
https://github.com/jarosm/KSP-BonVoyage/blob/81ba8c575d53e9475ee07ca906eb10b2b635f884/BonVoyage/RoverController.cs#L205-L260
https://github.com/jarosm/KSP-BonVoyage/blob/81ba8c575d53e9475ee07ca906eb10b2b635f884/BonVoyage/RoverController.cs#L315-L367
https://github.com/jarosm/KSP-BonVoyage/blob/81ba8c575d53e9475ee07ca906eb10b2b635f884/BonVoyage/RoverController.cs#L624-L652
https://github.com/jarosm/KSP-BonVoyage/blob/81ba8c575d53e9475ee07ca906eb10b2b635f884/BonVoyage/RoverController.cs#L725-L757

And instead use the Kerbalism API to check the Kerbalism background resource simulation state and start / stop the rover accordingly : 
https://github.com/Kerbalism/Kerbalism/blob/ecde971e9193baa6b6bc77ff5e92c754f6c29fa1/src/Kerbalism/System/API.cs#L445-L448

There are likely a lot of other places that need to have some special handling. In fact, it would likely be easier to make a completetely separate "KerbalismRoverController" PartModule.

**Comment by jarosm** [comment]
@gotmachine You are right, that it will be better to handle it by making separate classes. I have finally a time to work on it. Thanks for the heads up.
